### PR TITLE
Trival CI fix

### PR DESCRIPTION
**Pull request by Sean-StarLabs**

### CODE REVIEW

**Comment by Sean-StarLabs** [comment]
Only fixes kinetic, focal and jammy can't get the test firmware as it doesn't have `git`.

File: contrib/ci/ubuntu.sh [hunk 1]
```diff
@@ -2,19 +2,20 @@
...
+./contrib/ci/fwupd_setup_helpers.py install-dependencies --yes -o ubuntu
+
 #check meson is new enough
 ./contrib/ci/fwupd_setup_helpers.py test-meson
 
 #check markdown is new enough
 ./contrib/ci/fwupd_setup_helpers.py test-markdown
 
 #clone test firmware if necessary
+apt update -qq && apt install git -y
```
Comment (superm1) [inline contrib/ci/ubuntu.sh:18]: is this step actually necessary, or was it added before you made the other changes?

Comment (Sean-StarLabs) [reply contrib/ci/ubuntu.sh:18]: First 2 commits fix Kinetic
All 3 fix jammy
Focal fails a self test - think that's unrelated tho

Comment (Sean-StarLabs) [reply contrib/ci/ubuntu.sh:18]: Focal:
```
 6/35 fwupdplugin-self-test                  FAIL             0.19s   killed by signal 6 SIGABRT
>>> MALLOC_PERTURB_=141 G_TEST_BUILDDIR=/github/workspace/build/libfwupdplugin G_TEST_SRCDIR=/github/workspace/libfwupdplugin /github/workspace/build/libfwupdplugin/fwupdplugin-self-test
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
Listing only the last 100 lines from a long log.
# DEBUG: Disable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Disable
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: Disable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Disable
# DEBUG: Auto
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Auto
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: Disable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Disable
# DEBUG: Disable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Disable
# DEBUG: Yes
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Yes
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: Disable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Disable
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: Disable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Disable
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: Disable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Disable
# DEBUG: Disable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Disable
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: AHCI
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: AHCI
# DEBUG: Disable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Disable
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: Only DIAG Error Code
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Only DIAG Error Code
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# DEBUG: Disable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Disable
# DEBUG: 20 sec
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: 20 sec
# DEBUG: Auto
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Auto
# DEBUG: 1 - Lower Fan Speed
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: 1 - Lower Fan Speed
# DEBUG: No
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: No
# DEBUG: Low Speed
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Low Speed
# DEBUG: Enable
** (/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): DEBUG: 12:11:35.154: Enable
# GLib-DEBUG: setenv()/putenv() are not thread-safe and should not be used after threads are created
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): GLib-DEBUG: 12:11:35.154: setenv()/putenv() are not thread-safe and should not be used after threads are created
# FuBiosSettings-DEBUG: re-initializing attributes
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuBiosSettings-DEBUG: 12:11:35.154: re-initializing attributes
# FuBiosSettings-DEBUG: loaded 75 BIOS settings
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuBiosSettings-DEBUG: 12:11:35.157: loaded 75 BIOS settings
# GLib-DEBUG: setenv()/putenv() are not thread-safe and should not be used after threads are created
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): GLib-DEBUG: 12:11:35.157: setenv()/putenv() are not thread-safe and should not be used after threads are created
# FuBiosSettings-DEBUG: re-initializing attributes
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuBiosSettings-DEBUG: 12:11:35.157: re-initializing attributes
# FuBiosSettings-DEBUG: skipping non-directory /github/workspace/libfwupdplugin/tests/bios-attrs/dell-xps13-9310/strings.txt/attributes
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuBiosSettings-DEBUG: 12:11:35.157: skipping non-directory /github/workspace/libfwupdplugin/tests/bios-attrs/dell-xps13-9310/strings.txt/attributes
# FuBiosSettings-DEBUG: reset_bios is not supported
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuBiosSettings-DEBUG: 12:11:35.158: reset_bios is not supported
# FuBiosSettings-DEBUG: loaded 110 BIOS settings
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuBiosSettings-DEBUG: 12:11:35.160: loaded 110 BIOS settings
ok 10 /fwupd/bios-attrs{load}
# FuSecurityAttrs-DEBUG: duplicate security attr org.fwupd.hsi.PRX from plugin test implicitly obsoleted by plugin other-plugin
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuSecurityAttrs-DEBUG: 12:11:35.160: duplicate security attr org.fwupd.hsi.PRX from plugin test implicitly obsoleted by plugin other-plugin
# FuSecurityAttrs-DEBUG: security attr org.fwupd.hsi.PRX:other-plugin obsoleted by org.fwupd.hsi.PRX:test
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuSecurityAttrs-DEBUG: 12:11:35.160: security attr org.fwupd.hsi.PRX:other-plugin obsoleted by org.fwupd.hsi.PRX:test
# FuSecurityAttrs-DEBUG: security attr org.fwupd.hsi.PRX:other-plugin obsoleted by org.fwupd.hsi.PRX:test
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuSecurityAttrs-DEBUG: 12:11:35.160: security attr org.fwupd.hsi.PRX:other-plugin obsoleted by org.fwupd.hsi.PRX:test
# FuSecurityAttrs-DEBUG: security attr org.fwupd.hsi.PRX:test obsoleted by org.fwupd.hsi.BIOSGuard:test
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuSecurityAttrs-DEBUG: 12:11:35.160: security attr org.fwupd.hsi.PRX:test obsoleted by org.fwupd.hsi.BIOSGuard:test
# FuSecurityAttrs-DEBUG: security attr org.fwupd.hsi.PRX:other-plugin obsoleted by org.fwupd.hsi.BIOSGuard:test
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuSecurityAttrs-DEBUG: 12:11:35.160: security attr org.fwupd.hsi.PRX:other-plugin obsoleted by org.fwupd.hsi.BIOSGuard:test
ok 11 /fwupd/security-attrs{hsi}
# GLib-DEBUG: setenv()/putenv() are not thread-safe and should not be used after threads are created
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): GLib-DEBUG: 12:11:35.160: setenv()/putenv() are not thread-safe and should not be used after threads are created
# FuCommon-DEBUG: creating path /github/workspace/build/libfwupdplugin/tests/fwupd
(/github/workspace/build/libfwupdplugin/fwupdplugin-self-test:17533): FuCommon-DEBUG: 12:11:35.160: creating path /github/workspace/build/libfwupdplugin/tests/fwupd
Bail out! ERROR:../libfwupdplugin/fu-self-test.c:710:fu_plugin_config_func: assertion failed (error == NULL): did not write /github/workspace/build/libfwupdplugin/tests/fwupd/test.conf file (g-io-error-quark, 0)
stderr:
**
ERROR:../libfwupdplugin/fu-self-test.c:710:fu_plugin_config_func: assertion failed (error == NULL): did not write /github/workspace/build/libfwupdplugin/tests/fwupd/test.conf file (g-io-error-quark, 0)

```

Comment (hughsie) [reply contrib/ci/ubuntu.sh:18]: Is build/libfwupdplugin/tests/fwupd/test.conf not mutable for some reason?

Comment (superm1) [reply contrib/ci/ubuntu.sh:18]: > First 2 commits fix Kinetic
> All 3 fix jammy
> Focal fails a self test - think that's unrelated tho

I think we should explicitly add git to dependencies.xml (but not in control)

Comment (Sean-StarLabs) [reply contrib/ci/ubuntu.sh:18]: Seems to already be there?

Comment (superm1) [reply contrib/ci/ubuntu.sh:18]: If it's already there this is masking a different problem.

Comment (Sean-StarLabs) [reply contrib/ci/ubuntu.sh:18]: Seems so - debian:stable is now also failing, haven't seen the problem yet :/

Comment (superm1) [reply contrib/ci/ubuntu.sh:18]: I suspect your containers are going stale from what's on the repos.  If so; maybe what we're missing is an `apt update` call at the beginning of these jobs before dependencies are resolved.

File: contrib/ci/dependencies.xml [hunk 2]
```diff
@@ -494,6 +504,30 @@
       <package variant="x86_64" />
     </distro>
   </dependency>
+  <dependency id="libglib2.0-doc">
```
Comment (superm1) [inline contrib/ci/dependencies.xml:507]: this should not be needed; if it's we need to fix the reason it's needed

File: contrib/ci/ubuntu.sh [hunk 3]
```diff
@@ -2,19 +2,20 @@
...
 
-#check for and install missing dependencies
-./contrib/ci/fwupd_setup_helpers.py install-dependencies --yes -o ubuntu
-
 #check we have pip
 ./contrib/ci/fwupd_setup_helpers.py install-pip --yes -o ubuntu
 
+#check for and install missing dependencies
+./contrib/ci/fwupd_setup_helpers.py install-dependencies --yes -o ubuntu
+
```
Comment (superm1) [inline contrib/ci/ubuntu.sh:10]: I'm not convinced it's right to move this.  Pip isn't needed for install-dependencies command.

**Comment by hughsie** [comment]
@Sean-StarLabs are bits of this still needed?

**Pull request closed without merging**